pkg/platform/kube/functionres/lazy.go

```diff
@@ -1880,6 +1874,22 @@ func (lc *lazyClient) populateServiceSpec(ctx context.Context,
 			"ports", spec.Ports)
 	}
 
+	// add additional ports for sidecars
+	if function.Spec.Sidecars != nil {
+		if spec.Ports == nil || len(spec.Ports) == 0 {
+			spec.Ports = []v1.ServicePort{}
+		}
+		for _, sidecar := range function.Spec.Sidecars {
+			for _, port := range sidecar.Ports {
+				spec.Ports = append(spec.Ports, v1.ServicePort{
+					Name:       sidecar.Name,
+					Port:       port.ContainerPort,
+					TargetPort: intstr.FromInt(int(port.ContainerPort)),
+				})
+			}
```

Comment:
if this is called from createOrUpdate
you need to make sure the spec.ports list will not get duplicate values from previous instances